### PR TITLE
Md Book setup

### PR DESCRIPTION
**Pull request by TimonPost**
## Changes

- Quin introduction description (basicly an include of the README), this serves as the first entry page to the book. 

This serves as the landing page for the book which I think is nice to have. Even if it is just the readme. 

- Quick QUIC introduction; 

This is an initial, non-technical, short introduction to the QUIC protocol; I will expand on in-depth-subjects in later chapters. 
The goal is to elaborate on various aspects of QUIC while not trying to get to deep into any meat. What I want to accomplish here is to introduce the user to this protocol and shine a light on what is to come in the next sections of the book. 

I want to add other various key aspects of quic in relation to quin under this chapter.

- CI definition to build the md book

This PR is mainly focused on the setup. I will be creating PR's for each page.

## How to use

To view the book:

```
cd quinn/docs
cargo install mdbook
mdbook serve
```

## Related 

- #865 

## Needed
- @djc can you create the gh-pages branch and the GITHUB token as we spoke of in the gitter. 

https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-first-deployment-with-github_token

### CODE REVIEW

File: docs/src/quic.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+QUIC is a general-purpose, transport layer, network protocol, built on top of UDP.
+It is still an internet [draft][draft] undergoing standardization by the IETF.
+This indicates that it is not yet stabilized. 
+Quinn focuses on satisfying the latest draft but may lag some drafts behind. 
+The fact that it is a draft does not detract from the fact that QUIC is already used by more than half of 
+all connections from Chrome web browsers to Google's servers with increasing adoption in the overall market. 
+
+QUIC aims to be nearly equivalent to a TCP connection. With the goals to improve the performance of connection-oriented web applications, 
+reduce connection and transport latency, and estimate bandwidth for better congestion control. 
+While the intentions of QUIC were original web-oriented, it suits other areas like the game-networking industry very well.
```
[the remark this answers is not in the document]
Comment (TimonPost) [reply docs/src/quic.md:12]: Thanks, I will try to re-write it in a better-structured way.

[the remark this answers is not in the document]
Comment (TimonPost) [reply docs/src/quic.md]: Could you give it another go? I refactored a great deal of this text.

**Review by djc — APPROVED** [review]

**Comment by djc** [comment]
I still don't like the text very much, but I've just merged this to unblock your other work.

**Comment by TimonPost** [comment]
Thanks for merging. Unfortunate that it still is not what you want. If you have concrete feedback on that which you don't like I can improve it in later cycles. Anyway, I will revisit it probably more in the future. It's all an iteration.

**Pull request merged**